pkg/provider/aws/provider.go

```diff
@@ -74,37 +75,57 @@ func (p *Provider) converterFor(resourceType string) resourceconverter.ResourceC
 
 func NewProviders(ctx context.Context, cfg cfg.Provider, logger *zap.Logger) ([]types.Provider, error) {
 	logger.Info("Connecting to AWS account")
+
+	optFns := []func(options *config.LoadOptions) error{
+		config.WithDefaultsMode(aws.DefaultsModeCrossRegion),
+	}
+
 	if cfg.Profile != "" {
 		logger.Sugar().Infof("Using AWS profile '%v'", cfg.Profile)
+		optFns = append(optFns, config.WithSharedConfigProfile(cfg.Profile))
 	}
-	defaultConfig, err := config.LoadDefaultConfig(ctx,
-		config.WithSharedConfigProfile(cfg.Profile),
-		config.WithDefaultsMode(aws.DefaultsModeCrossRegion),
-	)
+	awsConfig, err := config.LoadDefaultConfig(ctx, optFns...)
 	if err != nil {
 		return nil, err
 	}
 
-	identity, err := awsutil.VerifyCreds(ctx, defaultConfig)
+	if cfg.RoleArn != "" {
+		logger.Sugar().Infof("Assuming AWS role '%v'", cfg.RoleArn)
+		creds, err := assumeRoleCredentials(awsConfig, cfg.RoleArn)
+		if err != nil {
+			return nil, err
+		}
+		optFns = append(optFns, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(creds.AccessKeyID, creds.SecretAccessKey, creds.SessionToken)))
```

Comment:
Is using [`config.WithAssumeRoleCredentialOptions`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#WithAssumeRoleCredentialOptions) an option here? It would be better to use that if possible, since it should handle credential expiration/refresh, and make it easier to support additional `sts:AssumeRole` options more easily (such as session name and source identity)